R/model.R

```diff
@@ -1,63 +1,235 @@
+##'   or `Inf`) should be used where the parameter has infinite domain
+##'   up or down.  Currently used to translate from a bounded to
+##'   unbounded space for HMC, but we might also use this for
+##'   reflecting proposals in MCMC too.  If not present (and if not
+##'   overridden by `domain`) we assume that the model is valid
+##'   everywhere (i.e., that all parameters are valid from `-Inf` to
+##'   `Inf`.
+##'
+##' * `direct_sample`: A function to sample directly from the
 ##'   parameter space, given an [mcstate_rng] object to sample from.
 ##'   In the case where a model returns a posterior (e.g., in Bayesian
 ##'   inference), this is assumed to be sampling from the prior.
 ##'   We'll use this for generating initial conditions for MCMC where
-##'   those are not given, and possibly other uses.
-##'
-##' @param density Compute the model density for a vector of parameter
-##'   values; this is the posterior probability in the case of
-##'   Bayesian inference, but it could be anything really.  Models can
-##'   return `-Inf` if things are impossible, and we'll try and cope
-##'   gracefully with that wherever possible.
-##'
-##' @param gradient Compute the gradient of `density` with respect to
-##'   the parameter vector; takes a parameter vector and returns a
-##'   vector the same length.  For efficiency, the model may want to
-##'   be stateful so that gradients can be efficiently calculated
-##'   after a density calculation, or density after gradient, where
-##'   these are called with the same parameters.
-##'
-##' @param domain Information on the parameter domain.  This is a two
-##'   column matrix with `length(parameters)` rows representing each
-##'   parameter.  The parameter minimum and maximum bounds are given
-##'   as the first and second column.  Infinite values (`-Inf` or
-##'   `Inf`) should be used where the parameter has infinite domain up
-##'   or down.  Currently used to translate from a bounded to
-##'   unbounded space for HMC, but we might also use this for
-##'   reflecting proposals in MCMC too.
+##'   those are not given, and possibly other uses.  If not given then
+##'   when using [mcstate_sample()] the user will have to provide a
+##'   vector of initial states.
+##'
+##' * `gradient`: A function to compute the gradient of `density` with
+##'   respect to the parameter vector; takes a parameter vector and
+##'   returns a vector the same length.  For efficiency, the model may
+##'   want to be stateful so that gradients can be efficiently
+##'   calculated after a density calculation, or density after
+##'   gradient, where these are called with the same parameters.  This
+##'   function is optional (and may not be well defined or possible to
+##'   define).
 ##'
-##' @return An object of class `mcstate_model`, which can be used with
-##'   a sampler.
+##' @title Create basic model
+##'
+##' @param model A list or environment with elements as described in
+##'   Details.
```

Comment:
where is "Details"? is this a roxygen2 thing that will reference the text above?

Reply:
Everything after the first linebreak is Details in the help rendering (or should be)